File: TypeScript/test/jest/mailServiceSpy.ts
import { IMailService } from "@/IMailService";

export default class mailServiceSpy implements IMailService {
  private _failedLoginUserName!: string;

  get failedLoginUserName() {
    return this._failedLoginUserName;
  }

  set failedLoginUserName(value: string) {
    this._failedLoginUserName = value;
  }

  sendLoginFailureMail(userName: string): void {
    this.failedLoginUserName = userName;
  }
  sendSystemFailureMail(errorType: number): void {}
}

File: CSharpCore/CSharpCoreTest/MailServiceSpy.cs
namespace CSharpCore.Test
{
    public class MailServiceSpy : IMailService
    {
        private string failedLoginUserName;

        public string FailedLoginUserName { get => failedLoginUserName; set => failedLoginUserName = value; }

        public void SendLoginFailureMail(string userName)
        {
            FailedLoginUserName = userName;
        }

        public void SendSystemFailureMail(int errorType)
        {

        }
    }
}